src/devices/LedDisplay/SevenSegmentFontHelper.cs

```diff
@@ -0,0 +1,162 @@
+            { 'J', Segment.TopRight | Segment.BottomRight | Segment.Bottom },
+            { 'l', Segment.BottomLeft | Segment.TopLeft },
+            { 'L', Segment.Bottom | Segment.BottomLeft | Segment.TopLeft },
+            { 'n', Segment.BottomRight | Segment.BottomLeft | Segment.Middle },
+            { 'N', Segment.Top | Segment.TopRight | Segment.BottomRight | Segment.BottomLeft | Segment.TopLeft },
+            { 'o', Segment.BottomRight | Segment.Bottom | Segment.BottomLeft | Segment.Middle },
+            { 'O', Segment.Top | Segment.TopRight | Segment.BottomRight | Segment.Bottom | Segment.BottomLeft | Segment.TopLeft },
+            { 'p', Segment.Top | Segment.TopRight | Segment.BottomLeft | Segment.TopLeft | Segment.Middle },
+            { 'P', Segment.Top | Segment.TopRight | Segment.BottomLeft | Segment.TopLeft | Segment.Middle },
+            { 'r', Segment.BottomLeft | Segment.Middle },
+            { 'R', Segment.Top | Segment.TopRight | Segment.BottomRight | Segment.BottomLeft | Segment.TopLeft | Segment.Middle },
+            { 's', Segment.Top | Segment.BottomRight | Segment.Bottom | Segment.TopLeft | Segment.Middle },
+            { 'S', Segment.Top | Segment.BottomRight | Segment.Bottom | Segment.TopLeft | Segment.Middle },
+            { 't', Segment.Bottom | Segment.BottomLeft | Segment.TopLeft | Segment.Middle },
+            { 'u', Segment.BottomRight | Segment.Bottom | Segment.BottomLeft },
+            { 'U', Segment.TopRight | Segment.BottomRight | Segment.Bottom | Segment.BottomLeft | Segment.TopLeft },
+            { 'y', Segment.TopRight | Segment.BottomRight | Segment.Bottom | Segment.TopLeft | Segment.Middle },
+            { 'Y', Segment.TopRight | Segment.BottomRight | Segment.Bottom | Segment.TopLeft | Segment.Middle },
+            { 'z', Segment.Top | Segment.TopRight | Segment.Bottom | Segment.BottomLeft | Segment.Middle },
+            { 'Z', Segment.Top | Segment.TopRight | Segment.Bottom | Segment.BottomLeft | Segment.Middle },
+            { '-', Segment.Middle },
+            { '_', Segment.Bottom },
+            { '|', Segment.BottomLeft | Segment.TopLeft },
+            { '°', Segment.Top | Segment.TopRight | Segment.TopLeft | Segment.Middle },
+            { ' ', Segment.None }
+        };
+
+        /// <summary>
+        /// Convert character value to corresponding font segments
+        /// </summary>
+        /// <param name="value">input character</param>
+        /// <returns>corresponding font bits</returns>
+        public static Segment GetCharacter(char value) => s_characterFontTable.ContainsKey(value) ? s_characterFontTable[value] : Segment.None;
+
+        /// <summary>
+        /// Convert a string of characters to corresponding font segments
+        /// </summary>
+        /// <param name="value">input string</param>
+        /// <returns>list of corresponding font segments</returns>
+        public static ReadOnlySpan<Segment> GetString(ReadOnlySpan<char> value)
```

Comment:
Returning span should only be done when span is subspan of the original span which is not the case here.

When array needs to be returned you have 2 options to do it correctly (or rather in way it makes sense):
- return array directly - wrapping with span doesn't give you anything in terms of allocations and it makes the code a bit harder
- return void and take output span as an input (recommended here):
`public static void GetString(ReadOnlySpan<char>, Span<Segment>);`

and then usage as following:

```csharp
string text = "abc";
Segment[] segments = new Segment[foo.Length];
XXX.GetString(text, segments);
```

Reply:
I'm still getting used to spans 🙂

From an interface usability standpoint I think returning an array is better.

Reply:
I don't mind either way since expected usage is that you update this relatively rarely (no more than few times per second)